haystack/components/embedders/openai_document_embedder.py

```diff
@@ -174,25 +177,32 @@ def _prepare_texts_to_embed(self, documents: List[Document]) -> List[str]:
 
             # copied from OpenAI embedding_utils (https://github.com/openai/openai-python/blob/main/openai/embeddings_utils.py)
             # replace newlines, which can negatively affect performance.
-            text_to_embed = text_to_embed.replace("\n", " ")
-            texts_to_embed.append(text_to_embed)
+            texts_to_embed[doc.id] = text_to_embed.replace("\n", " ")
         return texts_to_embed
 
-    def _embed_batch(self, texts_to_embed: List[str], batch_size: int) -> Tuple[List[List[float]], Dict[str, Any]]:
+    def _embed_batch(self, texts_to_embed: Dict[str, str], batch_size: int) -> Tuple[List[List[float]], Dict[str, Any]]:
         """
         Embed a list of texts in batches.
         """
 
         all_embeddings = []
         meta: Dict[str, Any] = {}
-        for i in tqdm(
-            range(0, len(texts_to_embed), batch_size), disable=not self.progress_bar, desc="Calculating embeddings"
+        for batch in tqdm(
+            batched(texts_to_embed.values(), batch_size), disable=not self.progress_bar, desc="Calculating embeddings"
         ):
-            batch = texts_to_embed[i : i + batch_size]
+            args: Dict[str, Any] = {"model": self.model, "input": [b[1] for b in batch]}
```

Comment:
```suggestion
            args: Dict[str, Any] = {"model": self.model, "input": list(batch)}
```

I'm pretty sure of this...

BTW, I noticed that integration tests are not running